### PR TITLE
Password policy authenticator for IS 7.0.0

### PR DESCRIPTION
**Pull request by sandushi**
## Purpose
Need to onboard the password policy connector to IS 7.0.0

## Changes from this PR
-Bump the osgi service version ranges to compatible with IS 7.0.0
- Bump the framework and governance versions
- Change the pwd-reset.jsp to honor the brandings and new layout components
- Improve the password reset enforcer and password change handler to honor the rule based password policy expiry as well
- Fix the failed unit test cases
- Move the common logic to utils class
- Add the password validation rules to the password reset page when isPasswordInputValidationEnabled is enabled. In default this is enabled. With this change the password reset page of the connector will be improved as below with better UX.

Screenshot from 2024-12-17 18-45-42

## Related issue
https://github.com/wso2/product-is/issues/21890

### CODE REVIEW

File: component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordPolicyUtils.java [hunk 1]
```diff
@@ -18,30 +18,47 @@
...
+import org.wso2.carbon.identity.policy.password.models.PasswordExpiryRuleAttributeEnum;
+import org.wso2.carbon.identity.role.v2.mgt.core.exception.IdentityRoleManagementException;
+import org.wso2.carbon.identity.role.v2.mgt.core.RoleManagementService;
+import org.wso2.carbon.identity.role.v2.mgt.core.model.RoleBasicInfo;
 import org.wso2.carbon.idp.mgt.IdentityProviderManagementException;
 import org.wso2.carbon.idp.mgt.IdentityProviderManager;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.common.Group;
+
+import java.util.*;
```
Comment (sahandilshan) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordPolicyUtils.java:53]: Remove * from imports

File: component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordPolicyUtils.java [hunk 2]
```diff
@@ -188,4 +205,273 @@ public static String convertWindowsFileTimeToUnixTime(String windowsFileTime) {
...
+            for (Property property : properties) {
+                if (StringUtils.startsWith(property.getName(), PASSWORD_EXPIRY_RULES_PREFIX) &&
+                        StringUtils.isNotEmpty(property.getValue())) {
+                    try {
+                        PasswordExpiryRule passwordExpiryRule = new PasswordExpiryRule(property.getValue());
+                        passwordExpiryRules.add(passwordExpiryRule);
+                    } catch (Exception e) {
+                        // Log and skip the rule if an error occurred while parsing the rule, without failing the
+                        // authentication flow.
+                        if (log.isDebugEnabled()) {
```
Comment (sahandilshan) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordPolicyUtils.java:243]: Do we really need these two logs here, can't we improve the error log or may be converting error log to a warn

File: component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordPolicyUtils.java (hunk 1 above)
Comment (sandushi) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordPolicyUtils.java]: resolved with https://github.com/wso2-extensions/identity-outbound-auth-passwordPolicy/pull/61/commits/a7e572e7ffc256388e0c5aa3856bd328af3a4c93

File: component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordPolicyUtils.java (hunk 2 above)
Comment (sandushi) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordPolicyUtils.java]: resolved with https://github.com/wso2-extensions/identity-outbound-auth-passwordPolicy/pull/61/commits/a7e572e7ffc256388e0c5aa3856bd328af3a4c93

**Review by piraveena — APPROVED** [review]

**Pull request merged**